src/elm/Page/PublicProfile.elm

```diff
@@ -0,0 +1,226 @@
+            ]
+            [ div [ style "grid-area" "avatar" ] [ Avatar.view ipfsUrl profile.avatar "w-20 h-20" ]
+            , div [ style "grid-area" "info" ] [ viewUserInfo userName email account ]
+            , div [ style "grid-area" "desc" ] [ viewUserDescription description ]
+            , div [ style "grid-area" "extra" ]
+                [ viewUserExtendedInfo
+                    [ ( "Locations", [ location ] )
+                    , ( "Interests", profile.interests )
+                    ]
+                ]
+            ]
+        ]
+
+
+viewUserBalance : Int -> Html msg
+viewUserBalance amount =
+    div
+        [ class "flex flex-col justify-between items-center py-4 w-full w-full bg-gray-100 h-40 rounded"
+        ]
+        [ span [ class "text-xs uppercase text-green" ] [ text "User Balance" ]
+        , span [ class "text-indigo-500 text-3xl font-medium" ] [ text (String.fromInt amount ++ " cr") ]
+        , button
+            [ style "width" "calc(100% - 16px)"
+            , class "bg-orange-300 uppercase text-sm font-medium text-white h-10 rounded-lg max-w-xs"
+            ]
+            [ text "Transfer CR" ]
+        ]
+
+
+viewUserExtendedInfo : List ( String, List String ) -> Html msg
+viewUserExtendedInfo data =
+    div
+        [ class "grid divide-y"
+        ]
+        (List.map
+            (\x ->
+                div
+                    [ class "grid grid-cols-2 grid-rows-1"
+                    , style "grid-template-areas" "'key value'"
+                    ]
+                    [ span
+                        [ class "text-xs py-2 leading-6"
+                        , style "grid-area" "key"
+                        ]
+                        [ text (Tuple.first x) ]
+                    , span
+                        [ class "text-indigo-500 font-medium text-xs text-right py-2 leading-6"
+                        , style "grid-area" "value"
+                        ]
+                        [ text (String.join ", " (Tuple.second x)) ]
+                    ]
+            )
+            data
+        )
+
+
+viewUserDescription : String -> Html msg
+viewUserDescription content =
+    p
+        [ class "text-sm text-gray-900" ]
+        [ text content ]
+
+
+viewUserInfo : String -> String -> String -> Html msg
+viewUserInfo name email username =
+    let
+        contentClasses =
+            [ "text-sm", "text-gray-900" ] |> String.join " "
+
+        headerClasses =
+            [ "font-medium", "text-2xl" ] |> String.join " "
+    in
+    ul []
+        [ li [ class headerClasses ]
+            [ text name ]
+        , li
+            [ class contentClasses ]
+            [ text email ]
+        , li
+            [ class contentClasses ]
+            [ text username ]
+        ]
+
+
+
+-- UPDATE
+
+
+update : Msg -> Model -> LoggedIn.Model -> UpdateResult
+update msg model loggedIn =
+    case msg of
+        CompletedProfileLoad (Ok Nothing) ->
+            -- TODO: not found account
```

Comment:
missing TODO here

Reply:
@lucca65 Is it possible to redirect the user to our 404 page in this case?

Reply:
It is possible to do redirects, yes, but instead we should just render a not found. Check the Page module